updated code with working functionalities.

rishabhjhs opened the pull request:
This code contains the working functionalities of the features asked to add on to the project

Commit:
updated code with working functionalities.

gouravnema commented on index.js:
```diff
@@ -109,18 +116,43 @@ function resetMoveInterval(){
+                                                                second argument is passed in reference with the speed so as to maintain the interval of 
+                                                                generated cars accordingly with speed*/                          
+    }
+else if(SPEED>=25 && SPEED<35)
```
Pls name your constants

gouravnema commented on index.js:
```diff
@@ -109,18 +116,43 @@ function resetMoveInterval(){
+      console.log(SPEED);
+      speedIntervalId = setInterval(frame, SPEED);
+      clearInterval(generatecarId);                               
+      if(SPEED>=35 && SPEED <=45){
```
pls extract crashlogic in a function

gouravnema commented on index.js:
```diff
@@ -109,18 +116,43 @@ function resetMoveInterval(){
+      speedIntervalId = setInterval(frame, SPEED);
+      clearInterval(generatecarId);                               
+      if(SPEED>=35 && SPEED <=45){
+      generatecarId=setInterval(genetateOponent,SPEED*100);      /*checking if the speed is in this particular range then the setInterval method is called 
```
pls fix typo though it was in existing code

gouravnema commented on index.js:
```diff
@@ -109,18 +116,43 @@ function resetMoveInterval(){
+    }
+else
+    {
+      generatecarId=setInterval(genetateOponent,SPEED*100);   // this condition is written so as to handle the event like if a user starts the game by a down
```
CamelCase

gouravnema commented on index.js:
```diff
@@ -130,11 +162,74 @@ function oponentMoveAnimation(el,lane){
+                reset_intervals();
+                flag=true; 
+                alert("GAME OVER");
+                window.location.reload();
```
lets not reload. Let's have a reset function which should just reset game state

gouravnema commented on index.js:
```diff
@@ -130,11 +162,74 @@ function oponentMoveAnimation(el,lane){
+
+           else if(car_currentPostion==21 && lane=="25vw" && pos==75)
+            {
+                console.log("crash occured");
```
can we not skip code repeat logic

gouravnema commented on index.js:
```diff
@@ -130,11 +162,74 @@ function oponentMoveAnimation(el,lane){
           cloneEl.style.left = lane;
           cloneEl.style.zIndex = Date.now() % 1000
-          if(pos >110){
+          var id_of_car = document.getElementById('player');
```
camel case

gouravnema commented on index.js:
```diff
@@ -130,11 +162,74 @@ function oponentMoveAnimation(el,lane){
-          if(pos >110){
+          var id_of_car = document.getElementById('player');
+          var car_currentPostion = parseInt(id_of_car.style.left || 1);    //here the by default value of car_currentposition is changed to 1 insted of 0
+         //console.log("car pos"+car_currentPostion);                     //to handle the crash event in case user just starts the game and do not move the car     
```
pls remove commented code

gouravnema commented on index.js:
```diff
@@ -130,11 +162,74 @@ function oponentMoveAnimation(el,lane){
+          //console.log(lane);
+          if(car_currentPostion==11 && lane=="15vw" && pos==75)          //These if else statements are written to handle the crash event if the above 
+          {                                                             //contition satisfy crash will occur and all the interval will close    
+                console.log("crash occured");
```
crash logic is only testing tail to head collision, it is not working on lateral collision.

Commit:
code updated with lateral crashing

this code contains lateral crashing logic and it also has all the functions which were asked to include.

gouravnema commented on index.js:
```diff
@@ -101,25 +128,49 @@ function carMoveAnimation(els, toPostion){
-    SPEED =speed;
-    console.log(SPEED);
-    speedIntervalId = setInterval(frame, SPEED);
+function resetSpeed(s)
```
nextSpeed instead of s

gouravnema commented on index.js:
```diff
@@ -130,18 +181,105 @@ function oponentMoveAnimation(el,lane){
+resetIntervals();
+}
+
+function crashOccurred(carCurrentPostion,lane,pos)
```
can we remove entire if else chain. try js objects (hash maps).

gouravnema commented on index.js:
```diff
@@ -130,18 +181,105 @@ function oponentMoveAnimation(el,lane){
+    else if(carCurrentPostion<=posSeven && lane==leftLane && (pos>=posSeventySeven && pos<=posNintyThree)) 
+        {       
+         flag=true; 
+         return true;
```
why setting and using flag(global var) when this function can return required boolean.

rishabhjhs commented:
Sir as per the logic i have written it actually checks all the conditions
that are needed to have a crash. If we use maps then also we have to get
the values from it and check the condition that is required to satisfy the
crash.

On Fri, 5 Oct 2018 at 11:35, Gourav Nema <notifications@github.com> wrote:

> *@gouravnema* commented on this pull request.
> ------------------------------
>
> In index.js
> <https://github.com/gouravnema/racer/pull/1#discussion_r222899396>:
>
> >  }
>
> +function resetGame(cloneEl)                        //this function is used to initialise the game
> +{
> +cloneEl.parentElement.removeChild(cloneEl);
> +resetSpeed(50);
> +flag=false;
> +resetIntervals();
> +}
> +
> +function crashOccurred(carCurrentPostion,lane,pos)
>
> can we remove entire if else chain. try js objects (hash maps).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/gouravnema/racer/pull/1#pullrequestreview-161898721>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AKS96ed0Znaz7EIfxrgGG76WI1ue2IuSks5uhva4gaJpZM4XHsFK>
> .
>

rishabhjhs commented:
because flag is used to perform an additional function as there are more
the 1 generated opponents and each opponent has its animation id  so to
stop them i have used flag that stops the other opponent apart from the
crashed one.

On Fri, 5 Oct 2018 at 11:37, Gourav Nema <notifications@github.com> wrote:

> *@gouravnema* commented on this pull request.
> ------------------------------
>
> In index.js
> <https://github.com/gouravnema/racer/pull/1#discussion_r222899609>:
>
> > +resetIntervals();
> +}
> +
> +function crashOccurred(carCurrentPostion,lane,pos)
> +{
> +    //These if else statements are written to handle the crash event if the above
> +    //contition satisfy crash will occur and all the interval will close
> +    if(carCurrentPostion==posEleven && lane==middleLane && pos==posSeventySeven)
> +        {
> +         flag=true;
> +          return true;
> +        }
> +    else if(carCurrentPostion<=posSeven && lane==leftLane && (pos>=posSeventySeven && pos<=posNintyThree))
> +        {
> +         flag=true;
> +         return true;
>
> why setting and using flag(global var) when this function can return
> required boolean.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/gouravnema/racer/pull/1#pullrequestreview-161898983>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AKS96UeHQbP38IslDTF7G1NaOgWd53UTks5uhvcfgaJpZM4XHsFK>
> .
>
